compiler/src/dotty/tools/dotc/config/Platform.scala

```diff
@@ -41,10 +36,12 @@ abstract class Platform {
   /** The given symbol is a method with the right name and signature to be a runnable program. */
   def isMainMethod(sym: SymDenotation)(implicit ctx: Context): Boolean
 
-  /** The given class has a main method. */
-  final def hasMainMethod(sym: Symbol)(implicit ctx: Context): Boolean =
+  /** The given class has a main method.
+   *  @param  staticOnly   only static main methods count
+   */
+  final def hasMainMethod(sym: Symbol, staticOnly: Boolean = false)(implicit ctx: Context): Boolean =
     sym.info.member(nme.main).hasAltWith {
-      case x: SymDenotation => isMainMethod(x)
+      case x: SymDenotation => isMainMethod(x) && (!staticOnly || x.isStatic)
```

Comment:
What's the benefit of having a staticOnly parameter instead of directly checking `sym.is(Module)` here ?

Reply:
Good point. We can just test directly.